pkg/logger/logger.go

```diff
@@ -0,0 +1,66 @@
+package logger
+
+import (
+	"os"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap/zapcore"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var logLevelMapping = map[string]int{
+	"devel":   0,
+	"default": 1, // default one when not set log-mode
+	"prod":    2,
+}
+
+// in each controller, to use different log level.
+func LogWithLevel(logger logr.Logger, level string) logr.Logger {
```

Comment:
will it check for "production" and "development" like doc says and ConfigLoggers does?

Reply:
i copy the line from "component" to the dsci level in README.
on one hand, i think good to have the same values on both places so user have a consistent experience.
on the other hand, it is a bit duplicated to have devel and development, prod and production here. 
Both are easy to change, but not sure which is better.

Reply:
> i copy the line from "component" to the dsci level in README.

Yeah, but does it mean that "devel" sets the verbosity level, but "development" doesn't since lookup in the map fails?

Reply:
yes, for the current code, it only support "" "devel" and "prod"
if we want to have "development" and "production", need to add these 2 into map.
or remove 2 lines in the README table.